src/Symfony/Component/Security/Http/Logout/LogoutUrlGenerator.php

```diff
@@ -151,6 +151,10 @@ private function getListener(?string $key): array
             }
         }
 
-        throw new \InvalidArgumentException('Unable to find the current firewall LogoutListener, please provide the provider key manually.');
+        if (null === $this->currentFirewallName) {
+            throw new \InvalidArgumentException('This request is not behind a firewall, please pass the firewall name manually to generate a logout URL.');
```

Comment:
I know I suggested these messages, but after reading them now in the code I think it's better to remove "please" here and below 🙂